test/e2e/self_hosted_test.go

```diff
@@ -64,10 +64,9 @@ var _ = Describe("When testing Cluster API working on single-node self-hosted cl
 			BootstrapClusterProxy:    bootstrapClusterProxy,
```

Comment:
As you were writing somewhere. We have two cases of this. The other one is where the ClusterClass doesn't have any MD/MP classes defined. Can we add an additional quickstart with clusterclass test case to cover the missing scenario?


(I think the one you already covered with this PR Is the more important/frequent one, so it's good to have more extensive coverage with the self-hosted test for this)